frontend/src/app/modules/imports/services/data.service.ts

```diff
@@ -17,21 +18,30 @@ import {
   ContentMappingValues,
 } from '../models/mapping.model';
 import { ConfigService } from '@geonature/services/config.service';
+import { ActivatedRoute } from '@angular/router';
 
 @Injectable()
 export class DataService {
   private urlApi = null;
+  private destination: string = null;
 
   constructor(
     private _http: HttpClient,
     public config: ConfigService
   ) {
-    this.urlApi = `${this.config.API_ENDPOINT}/import/synthese`;
-    // this.urlApi = `${this.config.API_ENDPOINT}/import/occhab`;
+    this.urlApi = `${this.config.API_ENDPOINT}/import`;
+  }
+
+  setDestination(destination: string) {
+    this.destination = destination;
+  }
```

Comment:
**Suggestion:** The `setDestination` function is used to set the `destination` property of the `DataService` class. However, it does not validate the input before setting it. This could potentially lead to incorrect values being set. You should add input validation to ensure the destination is a non-empty string. [enhancement]
```suggestion
  setDestination(destination: string) {
      if (!destination || typeof destination !== 'string') {
          throw new Error('Invalid destination');
      }
      this.destination = destination;
  }
```